packages/maskbook/src/plugins/NFT/UI/NFTInPost.tsx

```diff
@@ -0,0 +1,44 @@
+import { useAsync } from 'react-use'
+import Services from '../../../extension/service'
+import MaskbookPluginWrapper from '../../MaskbookPluginWrapper'
+import type { NFTInPostProps, TokenDetails } from '../types'
+import { parseNftUrl } from '../utils'
+import NFTCardUI from './NFTCardUI'
+import { openseaAssetsApiUrl } from '../constants'
+
+export default function NFTInPost(props: NFTInPostProps) {
+    const tokenDetailsResponse = useAsync(
+        async function () {
+            let updatedTokenDetails: TokenDetails = {}
+            const [address, tokenId] = parseNftUrl(props.nftUrl)
+
+            // if nftUrl is invalid
+            if (!(address && tokenId)) return updatedTokenDetails
+
+            const link = new URL(openseaAssetsApiUrl)
+            link.searchParams.set('asset_contract_address', address)
+            link.searchParams.set('token_ids', tokenId)
+
+            try {
+                const response = await Services.Helper.fetch(link.toString())
+                const { assets } = JSON.parse(await response.text())
+
+                // if `assets` is an empty array, finally block will return the empty `updatedTokenDetails`
+                if (assets[0].name) updatedTokenDetails.name = assets[0].name
```

Comment:
edge cases:

```ts
// 1. assets is undefined
assets = undefined
assets[0] // Uncaught TypeError: Cannot read property '0' of undefined
// 2. empty array
assets = []
assets[0] // always returns `undefined`
```

Reply:
This code is inside a try block so these edge cases will result in returning `null`.